plugin-ssh/che-plugin-ssh-ext-sshkey/src/main/java/org/eclipse/che/ide/ext/ssh/server/UserProfileSshKeyStore.java

```diff
@@ -114,89 +91,125 @@ public SshKeyPair genKeyPair(String host, String comment, String passPhrase) thr
 
     @Override
     public SshKeyPair genKeyPair(String host, String comment, String passPhrase, String keyMail) throws SshKeyStoreException {
-        User user = getUser(EnvironmentContext.getCurrent().getUser().getName());
+        ProfileDescriptor userProfile;
+        Map<String, String> keys;
         try {
-            Map<String, String> preferences = preferenceDao.getPreferences(user.getId());
-            if (keyMail == null) {
-                keyMail = user.getEmail() != null ? user.getEmail() : user.getId();
-            }
-            final String sshPrivateKeyAttributeName = sshKeyAttributeName(host, PRIVATE);
-            final String sshPublicKeyAttributeName = sshKeyAttributeName(host, PUBLIC);
-            // Be sure keys are not created yet.
-            if (preferences.get(sshPrivateKeyAttributeName) != null) {
-                throw new SshKeyStoreException(String.format("Private key for host: '%s' already exists. ", host));
-            }
-            if (preferences.get(sshPublicKeyAttributeName) != null) {
-                throw new SshKeyStoreException(String.format("Public key for host: '%s' already exists. ", host));
-            }
-            // Gen key pair.
-            KeyPair keyPair;
-            try {
-                keyPair = KeyPair.genKeyPair(genJsch, 2, 2048);
-            } catch (JSchException e) {
-                throw new SshKeyStoreException(e.getMessage(), e);
-            }
-            keyPair.setPassphrase(passPhrase);
-
-            SshKey privateKey;
-            SshKey publicKey;
-            ByteArrayOutputStream buff = new ByteArrayOutputStream();
-
-            keyPair.writePrivateKey(buff);
-            privateKey = new SshKey(sshPrivateKeyAttributeName, buff.toByteArray());
-            preferences.put(sshPrivateKeyAttributeName, new String(buff.toByteArray()));
+            userProfile = getUserProfile();
+            keys = getSshKeys();
+        } catch (ApiException | JsonParseException | IOException e) {
+            throw new SshKeyStoreException("Failed to generate keys for host '" + host + "'.", e);
+        }
+        if (keys == null || userProfile == null) {
```

Comment:
In line 67 you are not checking that keys is not null, but here you do. So what is correct behaviour?